bundles/framework/admin-announcements/view/AnnouncementsFormMapperHandler.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { announcementsCalls } from './AnnouncementsCalls';
+import { Message } from 'oskari-ui';
+
+/*
+Handler for admin-announcements forms.
+*/
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='admin-announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor (instance) {
+        super();
+        this.instance = instance;
+        this.sandbox = instance.getSandbox();
+        this.locale = instance.getLocalization();
+        this.announcementsCalls = announcementsCalls();
+        this.newTitle = 'Uusi ilmoitus';
+        this.state = {
+            update: false,
+            announcements: [],
+            title: 'Uusi Ilmoitus',
+            updated: false,
+            active: true,
+            activeKey: []
+
+        };
+    }
+
+    getAdminAnnouncements (callback) {
+        this.announcementsCalls.getAdminAnnouncements(function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
+                return;
+            }
+            callback(data);
+        });
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsCalls.saveAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+            }
+        });
+        this.updateState({
+            updated: false,
+            activeKey: []
+        });
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsCalls.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+            }
+        });
+        this.updateState({
+            updated: false,
+            activeKey: []
+        });
+    }
+
+    deleteAnnouncement (index, id) {
+        const test = { id };
+
+        // TODO: Better/different way to confirm deleting an announcement
+        if (window.confirm(this.instance.getLocalization('deleteAnnouncementConfirm'))) {
```

Comment:
You can wrap the delete button on an AntD confirm component. Then you don't need localization variable in the handler at all.